ui_testing/testcases/basic_tests/test08_contacts.py

```diff
@@ -0,0 +1,382 @@
+        first_contact_record = self.contact_page.get_random_contact_row()
+        self.contact_page.open_edit_page(row=first_contact_record)
+
+        self.base_selenium.LOGGER.info('acquire contact data to compare it after updating the persons')
+        contact_data = self.contact_page.get_full_contact_data()
+
+        self.base_selenium.LOGGER.info('Open contact persons page')
+        self.contact_page.get_contact_persons_page()
+        self.base_selenium.LOGGER.info('add new record to contact persons')
+        contact_persons_after_update = self.contact_page.create_update_contact_person(save=True)
+
+        self.base_selenium.LOGGER.info('Refresh to compare the data before and after refresh')
+        self.base_selenium.refresh()
+
+        contact_data_after_refresh = self.contact_page.get_full_contact_data()
+        if self.contact_page.compare_contact_main_data(data_after_save=contact_data_after_refresh, data_before_save=contact_data):
+            self.base_selenium.LOGGER.info('contact data have been saved successfully')
+        else:
+            self.base_selenium.LOGGER.info('contact data was not saved successfully, you should report a BUG')
+            self.assertEqual(True, False)
+
+        self.contact_page.get_contact_persons_page()
+        contact_persons_after_refresh = self.contact_page.get_contact_persons_data()
+
+        self.base_selenium.LOGGER.info('compare contact persons data after refresh')
+        if self.contact_page.compare_contact_persons_data(data_after_save=contact_persons_after_refresh, data_before_save=contact_persons_after_update):
+                self.base_selenium.LOGGER.info('contact persons have been saved successfully')
+        else:
+            self.base_selenium.LOGGER.info('contact persons was not saved successfully, you should report a BUG')
+            self.assertEqual(True, False)
+        
+
+    @skip('https://modeso.atlassian.net/browse/LIMS-6394')
+    def test_08_delete_contact_person(self):
+        """
+        Contact: Edit Approach: Make sure that you can delete any contact person from the edit mode 
+        LIMS-6387
+        """
+
+        self.base_selenium.LOGGER.info('in this case, we will delete contact person that exist in the first record in the table')
+        self.base_selenium.LOGGER.info('because in the sequence of test cases, first record will contain for sure a contact person')
+
+        self.base_selenium.LOGGER.info('select the first contact record')
+        first_contact_record = self.contact_page.result_table()[0]
+
+        self.base_selenium.LOGGER.info('open the record in edit form')
+        self.contact_page.open_edit_page(row=first_contact_record)
+
+        self.base_selenium.LOGGER.info('acquire the data from the form to compare it with the data after saving to make sure it is saved correctly')
+        contact_data = self.contact_page.get_full_contact_data()
```

Comment:
I dont knw, API calls can be better here?

Reply:
kinda the same